packages/react-next/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -1,210 +1,130 @@
-      );
-    }
-    return null;
-  }
+const getClassNames = classNamesFunction<IMessageBarStyleProps, IMessageBarStyles>();
 
-  private _getDismissSingleLine(): JSX.Element | null {
-    if (this.props.onDismiss) {
-      return <div className={this._classNames.dismissSingleLine}>{this._getDismissDiv()}</div>;
-    }
-    return null;
+const getAnnouncementPriority = (messageBarType: MessageBarType.info | MessageBarType): 'assertive' | 'polite' => {
+  switch (messageBarType) {
+    case MessageBarType.blocked:
+    case MessageBarType.error:
+    case MessageBarType.severeWarning:
+      return 'assertive';
   }
-
-  private _getExpandSingleLine(): JSX.Element | null {
-    if (!this.props.actions && this.props.truncated) {
-      return (
-        <div className={this._classNames.expandSingleLine}>
+  return 'polite';
+};
+
+export const MessageBarBase = React.forwardRef<HTMLDivElement, IMessageBarProps>((props, ref) => {
+  const [expandSingleLine, { toggle: toggleExpandSingleLine }] = useBoolean(false);
+  const labelId = useId('MessageBar');
+
+  const {
+    actions,
+    className,
+    children,
+    overflowButtonAriaLabel,
+    dismissIconProps,
+    styles,
+    theme,
+    messageBarType = MessageBarType.info,
+    onDismiss = undefined,
+    isMultiline = true,
+    truncated,
+    dismissButtonAriaLabel,
+    messageBarIconProps,
+  } = props;
+
+  const nativeProps = getNativeProps<React.HTMLAttributes<HTMLSpanElement>>(props, htmlElementProperties, [
+    'className',
+  ]);
+
+  const classNames: { [key in keyof IMessageBarStyles]: string } = getClassNames(styles, {
+    theme: theme!,
+    messageBarType: messageBarType || MessageBarType.info,
+    onDismiss: onDismiss !== undefined,
+    actions: actions !== undefined,
+    truncated: truncated,
+    isMultiline: isMultiline,
+    expandSingleLine: expandSingleLine,
+    className,
+  });
+
+  const iconProps = { iconName: expandSingleLine ? 'DoubleChevronUp' : 'DoubleChevronDown' };
+  const regionProps = actions || onDismiss ? { 'aria-describedby': labelId, role: 'region' } : {};
+  const actionsDiv = actions ? <div className={classNames.actions}>{actions}</div> : null;
+
+  return (
+    <div ref={ref} className={classNames.root} {...regionProps}>
+      <div className={classNames.content}>
+        <div className={classNames.iconContainer} aria-hidden>
+          {messageBarIconProps ? (
+            <Icon {...messageBarIconProps} className={css(classNames.icon, messageBarIconProps.className)} />
+          ) : (
+            <Icon iconName={ICON_MAP[messageBarType!]} className={classNames.icon} />
+          )}
+        </div>
+        <div className={classNames.text} id={labelId} role="status" aria-live={getAnnouncementPriority(messageBarType)}>
+          <span className={classNames.innerText} {...nativeProps}>
+            <DelayedRender>
+              <span>{children}</span>
+            </DelayedRender>
+          </span>
+        </div>
+        {!isMultiline &&
```

Comment:
I'm not sure some of the conditions and operator precedence are quite right. Also it would be clearer to do separate conditions for some parts.

```
// earlier
const dismissButton = onDismiss ? <IconButton .../> : null;

// here
{/* expand/collapse button */ !isMultiline && !actions && truncated && <div>......</div>}
{/* actions */ !isMultiline && actionsDiv}
{/* dismiss */ dismissButton && 
  (isMultiline ? dismissButton : <div className={classNames.dismissSingleLine}>{dismissButton}</div>)}
```